sdk/storage/Azure.Storage.Common/src/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -95,7 +95,7 @@ private static string BuildCanonicalizedHeaders(HttpMessage message)
             var sb = new StringBuilder();
             foreach (var headerName in
                 message.Request.Headers
-                .Select(h => h.Name)
+                .Select(h => h.Name.ToLowerInvariant())
                 .Where(name => name.StartsWith(Constants.HeaderNames.XMsPrefix, StringComparison.OrdinalIgnoreCase))
 #pragma warning disable CA1308 // Normalize strings to uppercase
                 .OrderBy(name => name.Trim().ToLowerInvariant()))
```

Comment:
But remove the `ToLowerInvariant` here so we're not the same operation twice.  Just order by name.